src/udev/udevadm-info.c

```diff
@@ -260,6 +265,117 @@ static int print_record(sd_device *device, const char *prefix) {
         return 0;
 }
 
+static int record_to_json(sd_device *device, JsonVariant **ret) {
+        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+        const char *str, *subsys;
+        dev_t devnum;
+        uint64_t q;
+        int i, ifi, r;
+
+        assert(device);
+        assert(ret);
+
+        assert_se(sd_device_get_devpath(device, &str) >= 0);
+        r = json_variant_set_field_string(&v, "P", str);
+        if (r < 0)
+                return r;
+
+        if (sd_device_get_sysname(device, &str) >= 0) {
+                r = json_variant_set_field_string(&v, "M", str);
+                if (r < 0)
+                        return r;
+        }
+
+        if (sd_device_get_sysnum(device, &str) >= 0) {
+                r = json_variant_set_field_string(&v, "R", str);
+                if (r < 0)
+                        return r;
+        }
+
+        if (sd_device_get_subsystem(device, &subsys) >= 0) {
+                r = json_variant_set_field_string(&v, "U", subsys);
+                if (r < 0)
+                        return r;
+        }
+
+        if (sd_device_get_devtype(device, &str) >= 0) {
+                r = json_variant_set_field_string(&v, "T", str);
+                if (r < 0)
+                        return r;
+        }
+
+        if (sd_device_get_devnum(device, &devnum) >= 0) {
+                _cleanup_free_ char *s = NULL;
+
+                if (asprintf(&s,
+                             "%c " DEVNUM_FORMAT_STR,
+                             streq_ptr(subsys, "block") ? 'b' : 'c',
+                             DEVNUM_FORMAT_VAL(devnum)) < 0)
+                        return -ENOMEM;
+
+                r = json_variant_set_field_string(&v, "D", str);
+                if (r < 0)
```

Comment:
if we do json let's do json, i.e. generate a sub object that consists of a string and two chars, i.e. `( "type" : "char", "major" : 4711, "minor" : 1234 )`